cmake/GenerateBuiltins.cmake

```diff
@@ -503,3 +503,117 @@ function (generate_builtins)
         ${DISPATCH_BUILTIN_CPP_FILES} ${COMMON_BUILTIN_CPP_FILES} ${TARGET_BUILTIN_CPP_FILES})
     add_dependencies(builtin builtins-cpp)
 endfunction()
+
+function (generate_generic_target_builtin_1 ispc_name target bit os)
+    set(include ${CMAKE_CURRENT_SOURCE_DIR}/stdlib/include)
+
+    string(REPLACE "-" "_" target_ ${target})
+    set(name "builtins_target_${target_}_${bit}bit_${os}")
+    set(input_ispc builtins/generic.ispc)
+
+    if (${target} MATCHES "x86_64")
+        set(arch "x86_64")
+    elseif (${target} MATCHES "aarch64")
+        set(arch "aarch64")
```

Comment:
I don't quite understand if generic targets are supposed to work on x86 and arm or not? Could you describe the logic regarding arch, please?

Reply:
I have removed arch from target names. We have now `generic-<mask_size>x<width>` with any --arch value.